monkey/common/concurrency/__init__.py

```diff
@@ -0,0 +1 @@
+from .protocols import Lock, Event
```

Comment:
This could also go in types?

Reply:
Types has a bunch of things that are general. These on the other hand are concurrency protocols and I think they have a good place.

Reply:
What do you mean by "general"? Why is this not general?

Reply:
These are `concurrency` protocols, I reckon we don't want to put them in general types and have one big file with all the types and protocols inside.